src/osd/PrimaryLogPG.cc

```diff
@@ -2044,10 +2044,19 @@ void PrimaryLogPG::do_op(OpRequestRef& op)
     }
   }
 
+  // check for op with rwordered and rebalance reads
+  if (m->has_flag(CEPH_OSD_FLAG_BALANCE_READS) &&
+      op->rwordered()) {
+    dout(4) << __func__ << ": rebalance reads with rwordered not allowed "
+       << *m << dendl;
+    osd->reply_op_error(op, -EINVAL);
```

Comment:
ACK.

Reply:
As the objecter no longer supports the problematic mixed ops, why are we checking instead of asserting? Is that for the sake of older clients?

Reply:
i just wanted to be sure: https://tracker.ceph.com/issues/70715#note-13 see the #2

Reply:
Yes, it's for old clients.
You basically have no control over client versions and you need to be ready to handle even misbehaving ones gracefully for long time (this fits particularly well to kernel clients).
Definitely you don't want to assert in such scenario.